src/test/javascript/spec/component/file-upload-exercise/file-upload-exercise-detail.component.spec.ts

```diff
@@ -31,6 +26,10 @@ import { NonProgrammingExerciseDetailCommonActionsComponent } from 'app/exercise
 import { ExerciseManagementStatisticsDto } from 'app/exercises/shared/statistics/exercise-management-statistics-dto';
 import { StatisticsService } from 'app/shared/statistics-graph/statistics.service';
 import * as sinon from 'sinon';
+import { AlertComponent } from 'app/shared/alert/alert.component';
+import { AlertErrorComponent } from 'app/shared/alert/alert-error.component';
+import { ExerciseDetailStatisticsComponent } from 'app/exercises/shared/statistics/exercise-detail-statistics.component';
+import { ExerciseDetailsComponent } from 'app/exercises/shared/exercise/exercise-details/exercise-details.component';
 
 chai.use(sinonChai);
 const expect = chai.expect;
```

Comment:
There are spies and stub used in this test, don't we need the `sinon.restore()` in an `afterEach()`?

Reply:
Yes we should use the restoring of the stubs 👍  Was not a problem because the spyOns always returned the same value. Adapted in 10c9c74